integrationTests/DoctrineDonationRepositoryTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace MatchBot\IntegrationTests;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Domain\DayOfMonth;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\DonationSequenceNumber;
+use MatchBot\Domain\DonationStatus;
+use MatchBot\Domain\DonorAccount;
+use MatchBot\Domain\DonorName;
+use MatchBot\Domain\EmailAddress;
+use MatchBot\Domain\Money;
+use MatchBot\Domain\PaymentMethodType;
+use MatchBot\Domain\PersonId;
+use MatchBot\Domain\Salesforce18Id;
+use MatchBot\Domain\StripeCustomerId;
+use MatchBot\Tests\TestCase;
+use Ramsey\Uuid\Uuid;
+
+class DoctrineDonationRepositoryTest extends IntegrationTest
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    public function testFindDonationsToSetPaymentIntent(): void
+    {
+        // arrange
+        $atDateTime = new \DateTimeImmutable('now');
+        $sut = $this->getService(DonationRepository::class);
+
+        $campaign = TestCase::someCampaign(
+            sfId: Salesforce18Id::ofCampaign('123456789012345678')
+        );
+
+        $em = $this->getService(EntityManagerInterface::class);
+
+        $mandate = new \MatchBot\Domain\RegularGivingMandate(
+            donorId: PersonId::of(Uuid::uuid4()->toString()),
+            donationAmount: Money::fromPoundsGBP(20),
+            campaignId: Salesforce18Id::ofCampaign($campaign->getSalesforceId()),
+            charityId: Salesforce18Id::ofCharity($campaign->getCharity()->getSalesforceId()),
+            giftAid: false,
+            dayOfMonth: DayOfMonth::of(2)
+        );
+        $donor = new DonorAccount(
+            uuid: $mandate->donorId,
+            emailAddress: EmailAddress::of('emailAddress@test.com'),
+            donorName: DonorName::of('donorFName-test', 'donorLName-test'),
+            stripeCustomerId: StripeCustomerId::of('cus_' . self::randomString())
+        );
+        $donor->setBillingCountryCode('GB');
+        $donor->setBillingPostcode('W1 5YU');
+        $mandate->activate($atDateTime);
+
+        $donation = $mandate->createPreAuthorizedDonation(
+            DonationSequenceNumber::of(2),
+            $donor,
+            $campaign
+        );
+
+        $donation->preAuthorize($atDateTime);
+
+        $em->persist($donor);
+        $em->persist($mandate);
+        $em->persist($campaign);
+        $em->persist($donation);
+        $em->flush();
+
+        $donations = $sut->findDonationsToSetPaymentIntent($atDateTime, 10);
+
+        $this->assertNotEmpty($donations);
```

Comment:
I know I suggested assertNotEmpty as replacement for asserting that the 0th item is non null, but we could be more specific than NotEmpty with
```suggestion
    $this->assertCount(1, $donations);
```

Also running on my local I'm currently getting an error "Failed asserting that an array is not empty". That's because the repository currently includes a check that the timestamp's day of month matches the dayOfMonth of the mandate - but the test is using 'now' as the timestamp as mentioned above so the third of the month, but the mandate is created with '2' as its day of month.

Reply:
resolved [here](https://github.com/thebiggive/matchbot/pull/1151/commits/6fd57be508b1274fe6bcfd40246efa4a663f363b)